src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp

```diff
@@ -34,388 +34,590 @@
-      if (!(boost::math::isnan)(mz_shift_next_peptide))
-      {
-        for (int isotope = 0; isotope < peaks_per_peptide_max_; ++isotope)
+        if (it_mz->getIntensity() > intensity_cutoff_)
         {
-          int mz_position = peptide * (peaks_per_peptide_max_ + 1) + isotope + 1; // index in m/z shift list
-          if (mz_shifts_actual[mz_position] >= mz_shift_next_peptide)
-          {
-            mz_shifts_actual[mz_position] = std::numeric_limits<double>::quiet_NaN();
-            mz_shifts_actual_indices[mz_position] = -1;
-          }
+          Peak1D peak;
+          peak.setMZ(it_mz->getMZ());
+          peak.setIntensity(it_mz->getIntensity());
+          spectrum.push_back(peak);
         }
       }
+      exp_picked_.addSpectrum(spectrum);
     }
-
-    // remove blacklisted peaks
-    // loop over isotopes in peptides
-    for (int isotope = 0; isotope < peaks_per_peptide_max_; ++isotope)
+    exp_picked_.updateRanges();
+    
+    // initialise blacklist blacklist_
+    blacklist_.reserve(exp_picked_.getNrSpectra());
+    // loop over spectra
+    for (MSExperiment::ConstIterator it_rt = exp_picked_.begin(); it_rt < exp_picked_.end(); ++it_rt)
     {
-      // loop over peptides
-      for (int peptide = 0; peptide < (int) pattern.getMassShiftCount(); ++peptide)
+      std::vector<BlacklistEntry> blacklist_spectrum;
+      blacklist_spectrum.reserve(it_rt->size());
+      // loop over m/z
+      for (MSSpectrum::ConstIterator it_mz = it_rt->begin(); it_mz < it_rt->end(); ++it_mz)
       {
-        int mz_position = peptide * (peaks_per_peptide_max_ + 1) + isotope + 1; // index in m/z shift list
-        int peak_index = mz_shifts_actual_indices[mz_position]; // index of the peak in the spectrum
-        if (peak_index != -1)
-        {
-          bool black = blacklist_[spectrum][peak_index].black;
-          bool black_exception = blacklist_[spectrum][peak_index].black_exception_mass_shift_index == pattern.getMassShiftIndex() &&
-                                 blacklist_[spectrum][peak_index].black_exception_charge == pattern.getCharge() &&
-                                 blacklist_[spectrum][peak_index].black_exception_mz_position == mz_position;
-          if (black && !black_exception)
-          {
-            mz_shifts_actual[mz_position] = std::numeric_limits<double>::quiet_NaN();
-            mz_shifts_actual_indices[mz_position] = -1;
-          }
-        }
+        BlacklistEntry entry = white;
+        blacklist_spectrum.push_back(entry);
       }
-    }
+      blacklist_.push_back(blacklist_spectrum);
+    }    
+  }
 
-    // count how many isotopic peaks seen simultaneously in all of the peptides
-    // and (optionally) remove peaks following missing ones
-    int peaks_found_in_all_peptides = peaks_per_peptide_max_;
-    for (int peptide = 0; peptide < (int) pattern.getMassShiftCount(); ++peptide)
+  MSExperiment MultiplexFiltering::getWhiteMSExperiment_(White2Original& mapping)
```

Comment:
use const-ref

Reply:
`mapping` is being returned, so cannot be `const`.